File: clothing-store-client/app/src/main/java/vn/clothing/store/networks/response/CartResponseModel.kt
package vn.clothing.store.networks.response

import com.google.gson.annotations.SerializedName
import vn.clothing.store.models.ProductDetails
import java.io.Serializable

class CartResponseModel {
    @SerializedName("cart_id")
    var cartId:String?=null
    @SerializedName("user_id")
    var userId:String?=null

    @SerializedName("list_item")
    var listItem:ArrayList<CartItemResponseModel>?=null


     class CartItemResponseModel : Serializable{
        var id:Int?=null
         @SerializedName("product_id")
        var productId:String?=null
        var quantity:Int?=null
        var color:String?=null
        var size:String?=null
        var image:String?=null
        var name:String?=null
        var price:Double?=null
    }
}

File: clothing_backend/models/response/CartResponseModel.js
// Định nghĩa và export từng class riêng lẻ

export class CartResponseModel {
    constructor(user = null, token = null) {
        this.user = user;
        this.token = token;
    }
}

export class CartItemResponseModel {
    constructor(id = null, productId = null, quantity = null, color = null, size = null, image = null, name = null, price = null) {
        this.id = id;
        this.productId = productId;
        this.quantity = quantity;
        this.color = color;
        this.size = size;
        this.image = image;
        this.name = name;
        this.price = price;
    }
}
